### PR TITLE
Update lookahead strategy

### PR DESCRIPTION
**Pull request by hxsz1997**
Update lookahead strategy. The new strategy is verified on ARC on OpenOrca/CNN-DailyMail datasets. The result is [here](https://github.com/analytics-zoo/nano/issues/1326)

### CODE REVIEW

File: python/llm/src/ipex_llm/transformers/lookup.py [hunk 1]
```diff
@@ -219,6 +232,8 @@ def lookup_generate(self,
 
     device_name = get_xpu_device_type(input_ids)
 
+    print(device_name, generation_config)
```
Comment (cyita) [inline python/llm/src/ipex_llm/transformers/lookup.py:235]: Please remove this line.

File: python/llm/src/ipex_llm/transformers/lookup.py [hunk 2]
```diff
@@ -190,18 +195,26 @@ def get_candidates(self,
...
 
         Args:
             num_matches (`int`):
                 The number of matches between the candidate sequences and the model predictions.
         """
-        if num_matches == self.num_output_tokens:
+        if self.num_output_tokens == 0:
+            ran = random.random() - 0.15
+            if ran <= accept_rate:
+                # print(f"random: {ran}, accept_rate: {accept_rate}")
```
Comment (cyita) [inline python/llm/src/ipex_llm/transformers/lookup.py:209]: Remove L209.

**Comment by cyita** [comment]
Please fix the style check.

**Comment by hxsz1997** [comment]
> Please fix the style check.

Sure. I have fix it.

**Pull request merged**